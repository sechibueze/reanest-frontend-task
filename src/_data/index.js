export const testData = {
  user: {
    name: "Sam John",
    id: {
      title: "ID",
      value: "3944949",
    },
    picture: "https://ibb.co/d21M4Ks",
  },
  wallet: {
    title: "Your Wallet",
    balance: {
      value: 9000,
      currency: "USD",
      title: "Wallet balance",
    },
  },
  loan: {
    title: "Are you cash strapped?",
    cta: {
      title: "Request for Salary Advance",
      link: "/salary-advance",
    },
  },
  billspayment: {
    title: "Bills Payment",
    items: [
      {
        id: "12",
        title: "Donations",
        icon: "fa",
      },
      {
        id: "13",
        title: "Airtime",
        icon: "fa-phone",
      },
      {
        id: "14",
        title: "Internet",
        icon: "fa",
      },
      {
        id: "15",
        title: "Data",
        icon: "fa",
      },
      {
        id: "22",
        title: "Electricity",
        icon: "fa",
      },
      {
        id: "72",
        title: "Betting",
        icon: "fa",
      },
    ],
  },
  transactions: {
    title: "Transactions",
    items: [
      {
        id: "12",
        image: "",
        title: "Glo ",
        status: "Success",
        category: "Airtime",
        date: "02/04/20",
        price: 100,
        currency: "USD",
      },
      {
        id: "12",
        title: "Spectranet",
        category: "Internet",
        date: "05/04/20",
        price: 1600,
        image: "",
        status: "Failed",
        currency: "USD",
      },
      {
        id: "12",
        title: "Airtel",
        category: "Data",
        date: "12/01/20",
        status: "Success",
        price: 1100,
        image: "",
        currency: "USD",
      },
    ],
  },
};
