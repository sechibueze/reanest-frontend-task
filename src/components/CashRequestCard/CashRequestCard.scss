@import "../../assets/theme/colors.scss";
.cash-request-card {
  background-color: #f1f1f1;

  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  border-radius: 0.5rem;
  width: 100%;
  align-self: stretch;

  & p {
    margin-bottom: 1.2rem;
    font-size: 0.7rem;
  }
}
.cash-request-cta {
  background-color: $blue;
  color: $white;
  text-align: center;
  padding: 0.5rem 0.9rem;
  font-size: 0.8rem;
}
