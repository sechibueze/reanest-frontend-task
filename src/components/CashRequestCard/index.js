import PropTypes from "prop-types";
import { testData } from "../../_data";
import "./CashRequestCard.scss";
const CashRequestCard = ({ cashRequestData }) => {
  return (
    <div className="cash-request-card">
      <p> {cashRequestData.title} </p>
      <a className="cash-request-cta" href={cashRequestData.cta.link}>
        {" "}
        {cashRequestData.cta.title}{" "}
      </a>
    </div>
  );
};
CashRequestCard.defaultProps = {
  cashRequestData: testData.loan,
};
CashRequestCard.propTypes = {
  cashRequestData: PropTypes.object.isRequired,
};
export default CashRequestCard;
