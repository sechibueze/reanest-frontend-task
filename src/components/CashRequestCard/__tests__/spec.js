import { shallow } from "enzyme";
import CashRequestCard from "../index";
import { testData } from "../../../_data";
describe("<CashRequestCard /> renders ", () => {
  const component = shallow(
    <CashRequestCard cashRequestData={testData.loan} />
  );
  const crCard = component.find(".cash-request-card");

  it("<CashRequestCard /> matches snapshot", () => {
    expect(component).toMatchSnapshot();
    expect(component.hasClass("cash-request-card")).toEqual(true);
  });
});
