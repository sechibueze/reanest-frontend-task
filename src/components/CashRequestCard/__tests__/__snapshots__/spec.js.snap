// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CashRequestCard /> renders  <CashRequestCard /> matches snapshot 1`] = `ShallowWrapper {}`;
